lib/pengine/unpack.c

```diff
@@ -1910,17 +1914,31 @@ process_rsc_state(resource_t * rsc, node_t * node,
             break;
         case action_fail_reset_remote:
             set_bit(rsc->flags, pe_rsc_failed);
-            tmpnode = NULL;
-            if (rsc->is_remote_node) {
-                tmpnode = pe_find_node(data_set->nodes, rsc->id);
+            if (is_set(data_set->flags, pe_flag_stonith_enabled)) {
+                tmpnode = NULL;
+                if (rsc->is_remote_node) {
+                    tmpnode = pe_find_node(data_set->nodes, rsc->id);
+                }
+                if (tmpnode &&
+                    is_baremetal_remote_node(tmpnode) &&
+                    tmpnode->details->remote_was_fenced == 0) {
```

Comment:
(Optimisation) I was thinking that instead of testing for 0, you could test for less than op->last_failure (or however one gets at it).  This would save you the pain (and maybe a few timing issues) of setting it back to 0 at TheRightTime(tm).